### PR TITLE
Add the ability to re-upload (community) resources, fix #194

### LINKED ISSUE #194: Add the ability to re-upload (community) resources
As a producer, I want to be able to re-upload (community) resources in order to propose a new file to reusers.

### PR DESCRIPTION
**Pull request by davidbgk**
Reupload resources + rename resources' fields (size and type to filesize and filetype) to avoid reserved keywords + upgrade to watai 0.7 + fix the title overriding on upload

### CODE REVIEW

**Comment by noirbizarre** [comment]
OK, ping me when it's ready

File: js/components/dataset/resource/file-form.vue [hunk 1]
```diff
@@ -102,11 +102,25 @@ export default {
...
+            } else {
+                endpoint += 'dataset_';
             }
-            return endpoint.urlify({dataset: this.dataset.id});
+            if (this.resource.id) {
+                endpoint += 'resource';
+                params.rid = this.resource.id;
+            } else {
+                endpoint += 'resources';
+            }
```
Comment (noirbizarre) [inline js/components/dataset/resource/file-form.vue]: How many endpoints/cases are there ?
Is'nt it possible to concat everything in a single if/else ?

File: js/models/communityresource.js [hunk 2]
```diff
@@ -11,7 +11,9 @@ export default class CommunityResource extends Model {
     fetch(ident) {
         ident = ident || this.id;
         if (ident) {
-            this.$api('datasets.retrieve_community_resource', {community_resource: ident}, this.on_fetched);
+            this.$api('datasets.retrieve_community_resource', {
```
Comment (noirbizarre) [inline js/models/communityresource.js]: Note for later: I think we should review the naming patterns to have the same all accross the application.

File: migrations/2015-10-08-migrate-resources-field-names.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+var datasets = db.dataset.find();
+var communityResources = db.community_resource.find();
+var counter = 0;
+var resourceCounter = 0;
+var communityResourceCounter = 0;
+
+datasets.map(function(dataset) {
+    if (dataset.resources) {
+        var resources = (dataset.resources).map(function(resource) {
```
Comment (noirbizarre) [inline migrations/2015-10-08-migrate-resources-field-names.js]: I think the parenthesis are useless here.

File: migrations/2015-10-08-migrate-resources-field-names.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+});
+print(counter, 'datasets modified.');
+print(resourceCounter, 'resources sizes modified.');
+
+communityResources.map(function(communityResource) {
+    communityResource.filetype = communityResource.type;
+    delete communityResource.type;
+    if (typeof communityResource.size !== undefined) {
+        communityResource.filesize = communityResource.size;
+        delete communityResource.size;
```
Comment (noirbizarre) [inline migrations/2015-10-08-migrate-resources-field-names.js]: I'm not sure if it's working in this case. You may have to do an `$unset` operation.

File: udata/core/dataset/api.py [hunk 5]
```diff
@@ -360,40 +399,38 @@ def get(self):
...
-        form.populate_obj(community_resource)
-        if not community_resource.organization:
-            community_resource.owner = current_user._get_current_object()
-        community_resource.modified = datetime.now()
-        community_resource.save()
-        return community_resource
+        form = api.validate(CommunityResourceForm, community)
+        form.populate_obj(community)
+        if not community.organization:
+            community.owner = current_user._get_current_object()
```
Comment (noirbizarre) [inline udata/core/dataset/api.py]: Is it necessary ? I think the `CurrentUserField` and the `PublishAsField` handle this.

File: udata/core/dataset/models.py [hunk 6]
```diff
@@ -398,6 +398,16 @@ def add_resource(self, resource):
         self.reload()
         post_save.send(self.__class__, document=self)
 
+    def update_resource(self, resource):
+        '''Perform an atomic update for an existing resource'''
+        index = [r.id for r in self.resources].index(resource.id)
```
Comment (noirbizarre) [inline udata/core/dataset/models.py]: Didn't know this one, nice ;)

**Comment by noirbizarre** [comment]
r+wc

File: js/components/dataset/resource/file-form.vue (hunk 1 above)
Comment (davidbgk) [inline js/components/dataset/resource/file-form.vue]: There are 4 cases from 2 conditions, I tried several approaches and couldn't figure out a better one but I'm open to suggestions. The issue with the current state of the code is that you have to follow the logic to recompose the endpoint which is far from being explicit…

File: js/models/communityresource.js (hunk 2 above)
Comment (davidbgk) [inline js/models/communityresource.js]: Agreed :)

File: migrations/2015-10-08-migrate-resources-field-names.js (hunk 3 above)
Comment (davidbgk) [inline migrations/2015-10-08-migrate-resources-field-names.js]: Last time I tried (a previous migration) it wasn't working.

File: migrations/2015-10-08-migrate-resources-field-names.js (hunk 4 above)
Comment (davidbgk) [inline migrations/2015-10-08-migrate-resources-field-names.js]: It looks to work locally. Did you try to run the migration?

File: udata/core/dataset/api.py (hunk 5 above)
Comment (davidbgk) [inline udata/core/dataset/api.py]: I thought too but it doesn't look to work as (I) expected :/ If I don't do it manually it's not updated auto-magically and I didn't find why.

File: udata/core/dataset/models.py (hunk 6 above)
Comment (davidbgk) [inline udata/core/dataset/models.py]: I'll try ;)

File: udata/core/dataset/api.py (hunk 5 above)
Comment (davidbgk) [inline udata/core/dataset/api.py]: Note that it was already the case, I just renamed `community_resource` to `community` in that part.

**Pull request merged**